hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/AWSClientConfig.java

```diff
@@ -378,17 +388,39 @@ private static void initSigner(Configuration conf,
    */
   private static void initRequestTimeout(Configuration conf,
       ClientOverrideConfiguration.Builder clientConfig) {
-    long requestTimeoutMillis = conf.getTimeDuration(REQUEST_TIMEOUT,
-        DEFAULT_REQUEST_TIMEOUT, TimeUnit.SECONDS, TimeUnit.MILLISECONDS);
+    final Duration requestTimeoutMillis = getDuration(conf, REQUEST_TIMEOUT,
+        DEFAULT_REQUEST_TIMEOUT, TimeUnit.SECONDS);
 
-    if (requestTimeoutMillis > Integer.MAX_VALUE) {
-      LOG.debug("Request timeout is too high({} ms). Setting to {} ms instead",
-          requestTimeoutMillis, Integer.MAX_VALUE);
-      requestTimeoutMillis = Integer.MAX_VALUE;
+    if (requestTimeoutMillis.toMillis() > 0) {
+      clientConfig.apiCallAttemptTimeout(requestTimeoutMillis);
     }
+  }
 
-    if(requestTimeoutMillis > 0) {
-      clientConfig.apiCallAttemptTimeout(Duration.ofMillis(requestTimeoutMillis));
+  /**
+   * Get duration. This may be negative; callers must check.
+   * If the config option is greater than {@code Integer.MAX_VALUE} milliseconds,
+   * it is set to that max.
+   * Logs the value for diagnostics.
+   * @param conf config
+   * @param name option name
+   * @param defVal default value
+   * @param defaultUnit unit of default value
+   * @return duration. may be negative.
```

Comment:
nit: `Get duration` -> `Get duration object representing value in milliseconds`?

Reply:
it's a java Duration class, so no need to specify unit